client/py_client/ipfs.py

```diff
@@ -29,3 +36,28 @@ def add_recursive(path_to_files):
     @staticmethod
     def add_recursive_multiple(paths):
         return [Ipfs.add_recursive(f) for f in paths]
+
+    @staticmethod
+    def add_recursive_remote(path_to_files, IPFS_API_KEY):
+        if IPFS_API_KEY != '':
+            ret = subprocess.run(
+                ["curl", "-X", "POST", "-F", f"file=@{path_to_files}", IPFS_API_KEY, "https://ipfs.infura.io:5001/api/v0/add"], stdout=subprocess.PIPE)
+        elif local_ipfs_key != None:
+            ret = subprocess.run(
+                ["curl", "-X", "POST", "-F", f"file=@{path_to_files}", local_ipfs_key, "https://ipfs.infura.io:5001/api/v0/add"], stdout=subprocess.PIPE)
+        # last line contains the directory cid
+        last = ret.stdout.splitlines()[-1]
```

Comment:
As the regex code is the same as mine, please extract it to a function and re-use it.

Reply:
i didn't manage this one yet, i uploaded a refactored method called refactoredCID, but if I refactor, I get an error message from the node: thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("invalid type: null, expected a string", line: 0, column: 0)', client/src/main.rs:384:26
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
I will try to solve it tomorrow morning

Reply:
now it works. eventhough the error message is not satisfying, the error was, that I wanted to create a new community with the gesell client and the ipfs_cid was None

Reply:
now the question would be: is it better to refactor the method to a static method or should I refactor to a method in the the helpers.py, which isn't part of the ipfs class, for example